modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/navigate.es.js

```diff
@@ -32,7 +32,7 @@ export default function (url, listeners) {
 			});
 		}
 	}
-	else {
+	else if (!url.startsWith('javascript:')) {
```

Comment:
Is this check enough? I imagine JS could be inserted in different exotic ways. For example, setting `foobar:alert(123)` opens an alert, to open `xdg-open` :man_shrugging: .

Probably a better and more comprehensive check would be to do the same as in [`canNavigate`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/app/App.js#L350-L381), just without the async-specific stuff. Something like:
```js
isValidURL(url) {
	try {
		url.startsWith('/')
			? new URL(url, window.location.origin)
			: new URL(url);

		return true;
	}
	catch (error) {
		return false;
	}
}
```